sympy/vector/implicitregion.py

```diff
@@ -76,37 +77,165 @@ def regular_point(self):
 
         >>> from sympy.abc import x, y
         >>> from sympy.vector import ImplicitRegion
-        >>> circle = ImplicitRegion((x, y), (x + 2)**2 + (y - 3)**2 - 16)
+        >>> circle = ImplicitRegion((x, y), x**2 + y**2 - 4)
         >>> circle.regular_point()
-        (-6, 3)
+        (2, 0)
+        >>> parabola = ImplicitRegion((x, y), x**2 - 4*y)
+        >>> parabola.regular_point()
+        (0, 0)
+        >>> r = ImplicitRegion((x, y, z), (x + y + z)**4)
+        >>> r.regular_point()
+        (-10, -10, 20)
 
-        """
+        Refrences
+        =========
+
+        - Erik Hillgarter, Rational Points on Conics.
 
-        # TODO: implement the algorithm to find regular point on a Conic.
-        # Now it iterates over a range and tries to find a point on the region.
+        """
         equation = self.equation
 
         if len(self.variables) == 1:
             return (list(solveset(equation, self.variables[0], domain=S.Reals))[0],)
         elif len(self.variables) == 2:
-            x, y = self.variables
 
-            for x_reg in range(-100, 100):
-                if not solveset(equation.subs(x, x_reg), self.variables[1], domain=S.Reals).is_empty:
-                    return (x_reg, list(solveset(equation.subs(x, x_reg), domain=S.Reals))[0])
-        elif len(self.variables) == 3:
+            if self.degree == 2:
+                coeffs = a, b, c, d, e, f = conic_coeff(self.variables, equation)
+
+                if b**2 == 4*a*c:
+                    x_reg, y_reg = self._regular_point_parabola(*coeffs)
+                    return x_reg, y_reg
+                else:
+                    x_reg, y_reg = self._regular_point_ellipse(*coeffs)
+                    return x_reg, y_reg
+
+        if len(self.variables) == 3:
             x, y, z = self.variables
 
             for x_reg in range(-10, 10):
                 for y_reg in range(-10, 10):
-                    if not solveset(equation.subs({x: x_reg, y: y_reg}), self.variables[2], domain=S.Reals).is_empty:
+                    if len(solveset(equation.subs({x: x_reg, y: y_reg}), self.variables[2], domain=S.Reals)) > 0:
```

Comment:
`solveset` returns a set... the previous expression looked better to me. Any reason for this change?